### PR TITLE
update range types api

### PR DESCRIPTION
**Pull request by sccolbert**
for discussion only at this point

### CODE REVIEW

**Comment by dwillmer** [comment]
LGTM

File: src/range/types.ts [hunk 1]
```diff
@@ -25,7 +24,7 @@ interface IInputRange<T> {
    * range is exhausted, since the length of a range may be infinite or
    * unknown.
    */
-  isEmpty(): boolean;
+  empty(): boolean;
```
Comment (blink1073) [inline src/range/types.ts]: `empty()` seems to close to clearing the range, I prefer `isEmpty()`.

**Comment by sccolbert** [comment]
This ended up being a bad idea. It forces 2x the number of range implementations per-container in order to support backward, vs 1 `backward` range utility function.

**Comment by sccolbert** [comment]
I'll make a few usable tweaks in a new PR.

**Pull request closed without merging**